app/controllers/saved_pages_controller.rb

```diff
@@ -0,0 +1,72 @@
+class SavedPagesController < ApplicationController
+  include AuthenticatedApiConcern
+
+  before_action :check_saved_page_exists, only: %i[destroy show]
+
+  # GET /api/saved_pages
+  def index
+    @saved_pages = SavedPage.where(oidc_user_id: user.id)
+    render_api_response(saved_pages: present_pages(@saved_pages))
+  end
+
+  # GET /api/saved_pages/:page_path
+  def show
+    render_api_response(saved_page: present_page(saved_page))
+  end
+
+  # POST /api/saved_pages
+  def create
+    saved_page = SavedPage.new(oidc_user_id: user.id, page_path: params[:page_path])
+
+    if saved_page.save
+      render_api_response(saved_page: present_page(saved_page))
+    else
+      handle_create_saved_page_errors(saved_page)
+    end
+  end
+
+  # DELETE /api/saved_pages
+  def destroy
+    return head :no_content if saved_page.destroy!
+
+    raise ApiError::SavedPageNotDestroyed, { page_path: params[:page_path] }
+  end
+
+private
+
+  def handle_create_saved_page_errors(saved_page)
+    error_types = saved_page.errors.map(&:type)
+
+    if error_types.include?(:taken)
+      raise ApiError::SavedPageDuplicate, { page_path: params[:page_path] }
+    end
+
+    if error_types.include?(:blank)
+      raise ApiError::SavedPagePathRequired, { page_path: params[:page_path] }
+    end
+
+    if error_types.any?
+      raise ApiError::SavedPageNotPersisted, { page_path: params[:page_path] }
+    end
+  end
+
+  def present_pages(saved_pages)
+    saved_pages.map { |saved_page| present_page(saved_page) }
+  end
+
+  def present_page(saved_page)
+    { "page_path" => saved_page.page_path }
+  end
+
+  def user
+    @user ||= OidcUser.find_by(sub: @govuk_account_session.user_id)
+  end
```

Comment:
Use `@govuk_account_session.user` instead, which handles creating the user record if it doesn't exist.

Reply:
nice!

Reply:
Done, great improvement thanks!